plugins/woocommerce/src/Admin/RemoteInboxNotifications/OptionRuleProcessor.php

```diff
@@ -25,7 +25,7 @@ public function process( $rule, $stored_state ) {
 		$default       = isset( $rule->default ) ? $rule->default : $default_value;
 		$option_value  = get_option( $rule->option_name, $default );
 
-		if ( $is_contains && ! is_array( $option_value ) ) {
+		if ( $is_contains && ! is_array( $option_value ) && ( ! is_string( $option_value ) || ! is_string( $rule->value ) ) ) {
```

Comment:
This additional condition is really confusing to me, maybe it would be worth adding a comment here, or is there a way to simplify this?
Why are we also checking if `rule->value` is not a string? Given the condition only sets the option value to an empty array.

Would `$is_contains && ! ( is_array( $option_value ) || is_string( $option_value ) )` be enough?

Reply:
Maybe it be worth writing some tests around this, to make sure the scenarios are covered.